codice/ddf
# DDF-2515 Updated catalog-core-commands that support the --cql option to also support temporal options

#### What does this PR do?
This PR updates catalog-core-commands to new karaf command annotations
and updates catalog-core-commands that support the `--cql` option to also support temporal options. 
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@peterhuffer @glenhein @mweser
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@millerw8
@kcwire
#### How should this be tested?

1. Confirm unit tests succeed.
2. Confirm that the command still works as expected and that the `--help` message is displayed correctly for all of the catalog commands:

  - `catalog:describe`
  - `catalog:dump`
  - `catalog:envlist`
  - `catalog:ingest`
  - `catalog:inspect`
  - `catalog:latest`
  - `catalog:migrate`
  - `catalog:range`
  - `catalog:remove`
  - `catalog:removeall`
  - `catalog:replicate`
  - `catalog:search`
  - `catalog:seed`
  - `catalog:spatial` (see [DDF-1282](https://jira.di2e.net/browse/DDF-1282))
  - `catalog:validate`

3. Confirm that the following commands include the `--cql` and temporal options and that the filtering options work as described, and try to use different temporal options:

  - `catalog:dump`
  - `catalog:migrate`
  - `catalog:remove`
  - `catalog:replicate`
  - `catalog:search`
  - `catalog:seed` (see https://github.com/codice/ddf/pull/1054)

#### Any background context you want to provide?
Related to https://github.com/codice/ddf/pull/1203.

When reviewing the code, it would probably help to look at code changes from each of the two commits separately so that file renames are recognized correctly.
#### What are the relevant tickets?
[DDF-2515](https://codice.atlassian.net/browse/DDF-2515)
#### Screenshots (if appropriate)
#### Checklist:
